src/parser/content-evaluator-module.ts

```diff
@@ -112,12 +110,59 @@ export class ContentEvaluatorModule implements Module {
     return commentsWithScore;
   }
 
+  _splitCommentsByPrompt(commentsWithScore: Readonly<GithubCommentScore>[]): {
+    commentsToEvaluate: CommentToEvaluate[];
+    reviewCommentsToEvaluate: ReviewCommentToEvaluate[];
+  } {
+    // exclude comments that have fixed relevance multiplier. e.g. review comments = 1
+    const commentsToEvaluate: CommentToEvaluate[] = [];
+    const reviewCommentsToEvaluate: ReviewCommentToEvaluate[] = [];
+    for (let i = 0; i < commentsWithScore.length; i++) {
+      const currentComment = commentsWithScore[i];
+      if (!this._fixedRelevances[currentComment.type]) {
+        if (currentComment.type & CommentKind.PULL) {
+          if (currentComment?.diff_hunk) {
```

Comment:
what happens with comments without code?

Reply:
They get default relevance of 1 from 
https://github.com/ubiquibot/conversation-rewards/blob/f56d997f5cd9b46a11af3797c7042e812b6e6a58/src/parser/content-evaluator-module.ts#L94

Reply:
That doesn't seem right.

Reply:
I have informed this https://github.com/ubiquibot/conversation-rewards/issues/45#issuecomment-2275548474
Many times, these comments are discussions or questions. Our prompt is focused on evaluating a code review comment, where code should be present. We can send them without code and treat them like code reviews.

Reply:
I'm not sure what the right answer is but I feel like maybe one more prompt could be the solution. We just need to provide the spec, full diff, the full pull conversation, and then ask if it's a useful comment I think.

Reply:
Full diff? There are 27 file changes in this issue we are talking. And it is not even a bigger issue at the moment in size. You want to send all these files to openai? That is a lot of openai tokens. 

Plus, these full files and conversations will be sent multiple times because that's how the plugin works. It sends a separate prompt for each user in an issue or pull request.

Two better solutions I see:
- set the relevance of such review comments with no code mention as 0, so that reviewers are forced to mention code to get rewarded. 
- second solution would be to send them as code review with diffHunk as empty. 

Also, setting some PR comment types as default relevance 1 wasn't wrong either. It is literally the main specification of this issue.

Reply:
I suppose it would be ideal to test each strategy to see which provides the most accurate results. I don't think setting relevance to 1 is the best solution compared to the other options.

Reply:
You are right. It would be nice to send everything so that things get evaluated correctly. The code files with PR roughly have about 0.56 million tokens, without the diff. It could cost more than 2.8$ for the code of this PR only. OpenAI batch API will reduce this cost to half but our current system works differently. So, this is a nice goal to have in the future if we can optimize the cost. Right now, I am going to send code review comments without diffHunk to OpenAi for evaluation with an empty diffHunk.

Reply:
I have updated the prompt to cover all comment types in the pull request. Normal PR comments and review comments with no code mention are sent with empty diffHunk. Code review comments with code mention have the code in the diffHunk.